### PR TITLE
TM-4326, update backend to handle TOD input

### PR DESCRIPTION
**Pull request by M00staff**
Ready for review!

Still need to test this in dev

Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2559)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/366)

### CODE REVIEW

File: talentmap_api/fsbid/services/agenda.py [hunk 1]
```diff
@@ -220,6 +220,8 @@ def convert_agenda_item_query(query):
 
 
 def fsbid_single_agenda_item_to_talentmap_single_agenda_item(data):
+    print("fsbid_single_agenda_item ----------> talentmap_single_agenda_item")
+    print(data)
```
Comment (M00staff) [inline talentmap_api/fsbid/services/agenda.py:224]: left my print statements for easy troubleshooting

File: talentmap_api/fsbid/services/agenda.py [hunk 2]
```diff
@@ -323,7 +326,10 @@ def map_tf(tf=None):
         "org": pydash.get(data, "agendaLegPosition[0].posorgshortdesc", None),
         "eta": pydash.get(data, "ailetadate", None),
         "ted": pydash.get(data, "ailetdtedsepdate", None),
-        "tod": pydash.get(data, "ailtodothertext", None),
+        # "tod": pydash.get(data, "ailtodothertext", None),
```
Comment (M00staff) [inline talentmap_api/fsbid/services/agenda.py:329]: OK - I'm like 80% sure this was using the wrong variable 

I still need to do multiple tests in dev1 - but, my understanding so far is that `ailtodothertext` is ONLY used when the 'OTHER' option is also selected

File: talentmap_api/fsbid/services/agenda.py [hunk 3]
```diff
@@ -384,7 +391,8 @@ def fsbid_aia_to_talentmap_aia(data):
         "org": pydash.get(data, "position[0].posorgshortdesc", None),
         "eta": pydash.get(data, "asgdetadate", None),
         "ted": pydash.get(data, "asgdetdteddate", None),
-        "tod": pydash.get(data, 'todshortdesc', None),
+        # "tod": pydash.get(data, 'todshortdesc', None),
+        "tod": pydash.get(data, 'asgdtoddesctext', None),
```
Comment (M00staff) [inline talentmap_api/fsbid/services/agenda.py:395]: this change was a guess - we mentioned the `asgd` acronym meant assignment detail or agenda leg assignment 

this is called when grabbing the legs from FSBID - so i am not sure what is correct, here

File: talentmap_api/fsbid/services/agenda.py [hunk 4]
```diff
@@ -483,9 +493,10 @@ def convert_agenda_item_leg_query(query, leg={}):
         "ailempseqnbr": int(pydash.get(query, "personId") or 0) or None,
         "ailperdetseqnum": int(pydash.get(query, "personDetailId") or 0) or None,
         "ailposseqnum": int(pydash.get(leg, "posSeqNum") or 0) or None,
-        "ailtodcode": pydash.get(leg, "tourOfDutyCode", ""),
-        "ailtodmonthsnum": None,
-        "ailtodothertext": None,
+        # "ailtodcode": pydash.get(leg, "tourOfDutyCode", ""),
+        "ailtodcode": pydash.get(leg, "tourOfDutyCode", None),
+        "ailtodmonthsnum": pydash.get(leg, "tourOfDutyMonths", None),
+        "ailtodothertext": pydash.get(leg, "tourOfDutyOtherText", None),
```
Comment (M00staff) [inline talentmap_api/fsbid/services/agenda.py:499]: these are the new variables the frontend will pass back - to be submitted... i still need to double check this in Dev

File: talentmap_api/fsbid/services/reference.py [hunk 5]
```diff
@@ -102,12 +102,11 @@ def fsbid_languages_to_talentmap_languages(data):
...
-        "code": data.get("code", 0),
-        "is_active": True,
-        "months": None,
-        "long_description": data.get("long_desc", None),
-        "short_description": data.get("long_desc", None)
+        "code": data.get("todcode", 0),
+        "is_active": True, # what is this?
+        "months": data.get("todmonthsnum", 0),
+        "long_description": data.get("toddesctext", None),
+        "short_description": data.get("todshortdesc", None)
```
Comment (M00staff) [inline talentmap_api/fsbid/services/reference.py:109]: just the updated mock data

File: talentmap_api/fsbid/urls/reference.py [hunk 6]
```diff
@@ -13,6 +13,7 @@
     url(r'^grades/$', views.FSBidGradesView.as_view(), name='FSBid-grades'),
     url(r'^languages/$', views.FSBidLanguagesView.as_view(), name='FSBid-languages'),
     url(r'^tourofduties/$', views.FSBidTourOfDutiesView.as_view(), name='FSBid-tour-of-duties'),
+    url(r'^toursofduty/$', views.FSBidToursOfDutyView.as_view(), name='FSBid-tours-of-duty'),
```
Comment (M00staff) [inline talentmap_api/fsbid/urls/reference.py:16]: Turns out there is a post/tourofduties endpoint
AND a references/tours-of-duty endpoint.

File: talentmap_api/fsbid/views/reference.py [hunk 7]
```diff
@@ -46,6 +46,11 @@ class FSBidTourOfDutiesView(BaseView):
     mapping_function = services.fsbid_tour_of_duties_to_talentmap_tour_of_duties
 
 
+class FSBidToursOfDutyView(BaseView):
+    uri = "v1/references/tours-of-duty"
+    mapping_function = services.fsbid_tours_of_duty_to_talentmap_tours_of_duty
```
Comment (M00staff) [inline talentmap_api/fsbid/views/reference.py:51]: I'm very upset about this turn of events

File: talentmap_api/fsbid/services/agenda.py [hunk 8]
```diff
@@ -384,7 +385,7 @@ def fsbid_aia_to_talentmap_aia(data):
         "org": pydash.get(data, "position[0].posorgshortdesc", None),
         "eta": pydash.get(data, "asgdetadate", None),
         "ted": pydash.get(data, "asgdetdteddate", None),
-        "tod": pydash.get(data, 'todshortdesc', None),
+        "tod": pydash.get(data, 'todcode', None),
```
Comment (M00staff) [inline talentmap_api/fsbid/services/agenda.py:388]: using the code here to stay consistent - frontend now expects the code, and will use the new reference endpoint to get the rest of the data
image

File: talentmap_api/fsbid/urls/reference.py (hunk 6 above)
Comment (scott062) [reply talentmap_api/fsbid/urls/reference.py:16]: Same idea from mock - we'll maintain both until new one is confirmed and then cutover OTHER existing functionality in a cleanup PR

File: talentmap_api/fsbid/services/reference.py [hunk 9]
```diff
@@ -102,12 +102,19 @@ def fsbid_languages_to_talentmap_languages(data):
 @staticmethod
 def fsbid_tour_of_duties_to_talentmap_tour_of_duties(data):
     return {
-        "id": data.get("code", 0),
         "code": data.get("code", 0),
-        "is_active": True,
-        "months": None,
         "long_description": data.get("long_desc", None),
-        "short_description": data.get("long_desc", None)
```
Comment (scott062) [inline talentmap_api/fsbid/services/reference.py:110]: Wonder if that data was coming through at one point and something went amiss and nobody caught it

**Review by scott062 — APPROVED** [review]

**Review by scott062 — APPROVED** [review]

File: talentmap_api/fsbid/services/reference.py (hunk 9 above)
Comment (M00staff) [reply talentmap_api/fsbid/services/reference.py]: yea - parsing through all of this and figuring out what was what was pretty tricky. 

went ahead and reverted this change here and in the mock - so the old TOD endpoint isn't changed in any way

**Pull request merged**